### PR TITLE
Fix: Update quick-error crate version from 1.2.3 to 2.0.1 in src/uu/cp

### LINKED ISSUE #2941: Update the quick-error crate to 2
```
src/uu/cp/Cargo.toml:quick-error = "1.2.3"
```
https://deps.rs/crate/itertools/0.8.2

we have already pr using 2
```
src/uu/pr/Cargo.toml:quick-error = "2.0.1"
```

### PR DESCRIPTION
**Pull request by Narasimha1997**
Hi, this PR fixes #2941 
This is my first contribution to this project. Starting small. :)

Thanks

### CODE REVIEW

**Comment by Narasimha1997** [comment]
Did the test cases fail because of dependency upgrade?

**Comment by tertsdiepraam** [comment]
It does indeed seem to be because of the change:
```
error: no rules expected the token `ERROR_CHECK`
   --> src/uu/cp/src/cp.rs:63:1
    |
63  | / quick_error! {
64  | |     #[derive(Debug)]
65  | |     pub enum Error {
66  | |         /// Simple io::Error wrapper
...   |
107 | |     }
108 | | }
    | |_^ no rules expected this token in macro call
    |
    = note: this error originates in the macro `quick_error` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `TUPLE`
   --> src/uu/cp/src/cp.rs:63:1
    |
63  | / quick_error! {
64  | |     #[derive(Debug)]
65  | |     pub enum Error {
66  | |         /// Simple io::Error wrapper
...   |
107 | |     }
108 | | }
    | |_^ no rules expected this token in macro call
    |
    = note: this error originates in the macro `quick_error` (in Nightly builds, run with -Z macro-backtrace for more info)
```

It might be because of one of these breaking changes from quick_error 2.0:
> 1. `cause` is removed, use `source`
> 2. No `description()` any more
> 3. Compiler warnings of edition 2018 eliminated

**Comment by Narasimha1997** [comment]
@tertsdiepraam 
Okay, In that case why aren't they failing in other two places where the similiar version of the create is being used?

**Comment by tertsdiepraam** [comment]
Because in those cases the code was adapted to the new version of quick_error. The `pr` util, for instance, does not use `cause`, but only `source`.

**Comment by Narasimha1997** [comment]
Cool this PR can be closed in that case

**Comment by tertsdiepraam** [comment]
It's probably just changing `cause` to `source`. Do you not want to give it a try?

**Comment by Narasimha1997** [comment]
Let me try in that case

**Comment by Narasimha1997** [comment]
Should work fine now with `quick-errors=2.0.1`, I have updated the `cp` crate to use `source` instead of `cause`.
Have tested this locally using:
```
cargo build --features cp
```

File: src/uu/cp/src/cp.rs [hunk 1]
```diff
@@ -953,7 +953,7 @@ fn adjust_canonicalization(p: &Path) -> Cow<Path> {
 /// contents to `target`.
 ///
 /// Any errors encountered copying files in the tree will be logged but
-/// will not cause a short-circuit.
+/// will not source a short-circuit.
```
Comment (sylvestre) [inline src/uu/cp/src/cp.rs:956]:
```suggestion
/// will not cause a short-circuit.
```

Comment (Narasimha1997) [reply src/uu/cp/src/cp.rs]: Fixed this. And I've rebased all my commits to one.

**Comment by Narasimha1997** [comment]
@sylvestre can this be merged??

**Comment by tertsdiepraam** [comment]
We can't be online all the time. We'll get to it when we can

**Comment by Narasimha1997** [comment]
Hey no no, I didn't ask in that way. There is a merge conflict so I asked haha. Sorry!

**Comment by tertsdiepraam** [comment]
I see, sorry! I fixed the conflict.

**Comment by tertsdiepraam** [comment]
Looks good! Thanks!

**Pull request merged**